app/helpers/layout_helper.rb

```diff
@@ -317,10 +317,11 @@ def alert_close(data_dismiss = 'alert')
     "<button type='button' class='close' data-dismiss='#{data_dismiss}' aria-hidden='true'>&times;</button>".html_safe
   end
 
-  def trunc_with_tooltip(text, length = 32)
-    text    = text.to_s
-    options = text.size > length ? { :'data-original-title' => text, :rel => 'twipsy' } : {}
-    content_tag(:span, truncate(text, :length => length), options).html_safe
+  def trunc_with_tooltip(text, length = 32, tooltip_text = "", shorten = true)
+    text = text.to_s.empty? ? tooltip_text.to_s : text.to_s
+    tooltip_text = tooltip_text.to_s.empty? ? text : tooltip_text.to_s
+    options = shorten && (text.size < length) ? {} : { :'data-original-title' => tooltip_text, :rel => 'twipsy' }
+    content_tag(:span, truncate(_(text), :length => length), options).html_safe
```

Comment:
I think this is still a problem, as per my previous comment.. if you call _(text) inside this helper, then you're either going to be:

a) assuming all text is untranslated and translating it - which if we pass in a subnet name (e.g. app/views/subnets/index.html.erb) in that happens to be in our translations, could cause it to be translated when it shouldn't be
b) translating something twice, if we passed in an already-translated value

I think this needs to stay on the *outside* of the helper.